components/cam/src/physics/cam/shoc.F90

```diff
@@ -1147,6 +1137,84 @@ subroutine diag_third_shoc_moments(&
+          zt_grid,zi_grid, isotropy_zi,&      ! Input
+          brunt_zi,w_sec_zi,thetal_zi,&       ! Input
+          wthv_sec_zi,shoc_mix_zi,&           ! Input
+          w3)                                 ! Output
+
+
+
+  ! perform clipping to prevent unrealistically large values from occuring
+  call clipping_diag_third_shoc_momnets(&
+          nlevi,shcol,w_sec_zi,&    !Input 
+          w3)                       !Output
+  
+  return
+
+end subroutine diag_third_shoc_moments
+
+subroutine compute_diag_third_shoc_moment(&
+          shcol,nlev,nlevi, &                 ! Input
+          w_sec,thl_sec, qw_sec, qwthl_sec,&  ! Input
+          wthl_sec, tke, dz_zt, dz_zi,&       ! Input
+          zt_grid,zi_grid, isotropy_zi,&      ! Input
+          brunt_zi,w_sec_zi,thetal_zi,&       ! Input
+          wthv_sec_zi,shoc_mix_zi,&           ! Input
+          w3)                                 ! Output
+
+  implicit none
+! INPUT VARIABLES
+  ! number of SHOC columns
+  integer, intent(in) :: shcol
+  ! number of midpoint levels
+  integer, intent(in) :: nlev
+  ! number of interface levels
+  integer, intent(in) :: nlevi
+  ! second order vertical velocity [m2/s2]
+  real(rtype), intent(in) :: w_sec(shcol,nlev)
+  ! second order liquid wat. potential temperature [K^2]
+  real(rtype), intent(in) :: thl_sec(shcol,nlevi)
+  ! second order total water mixing ratio [kg2/kg2]
+  real(rtype), intent(in) :: qw_sec(shcol,nlevi)
+  ! covariance of temp and moisture [K kg/kg]
+  real(rtype), intent(in) :: qwthl_sec(shcol,nlevi)
+  ! vertical flux of heat [K m/s]
+  real(rtype), intent(in) :: wthl_sec(shcol,nlevi)
+  ! turbulent kinetic energy [m2/s2]
+  real(rtype), intent(in) :: tke(shcol,nlev)
+  ! thickness centered on thermodynamic grid [m]
+  real(rtype), intent(in) :: dz_zt(shcol,nlev)
+  ! thickness centered on interface grid [m]
+  real(rtype), intent(in) :: dz_zi(shcol,nlevi)
+  ! heights of thermodynamics points [m]
+  real(rtype), intent(in) :: zt_grid(shcol,nlev)
+  ! heights of interface points [m]
+  real(rtype), intent(in) :: zi_grid(shcol,nlevi)
+
+  !Interpolated varaibles  
+  real(rtype), intent(in) :: isotropy_zi(shcol,nlevi)
+  real(rtype), intent(in) :: brunt_zi(shcol,nlevi)
+  real(rtype), intent(in) :: w_sec_zi(shcol,nlevi)
+  real(rtype), intent(in) :: thetal_zi(shcol,nlevi)
+  real(rtype), intent(in) :: wthv_sec_zi(shcol,nlevi)
+  real(rtype), intent(in) :: shoc_mix_zi(shcol,nlevi)
+  ! third moment of vertical velocity
+  real(rtype), intent(out) :: w3(shcol,nlevi)
+! LOCAL VARIABLES
+  integer i, j, k, kb, kc
+  real(rtype) :: omega0, omega1, omega2
+  real(rtype) :: X0, Y0, X1, Y1, AA0, AA1
+  real(rtype) :: zvar, x5var, iso, thedz, thedz2
+  real(rtype) :: theterm, cond, tsign
+  real(rtype) :: isosqrt, dthl2, dwthl, dtke, dw2, aw2
+  real(rtype) :: a0, a1, a2, a3, a4, a5
+  real(rtype) :: buoy_sgs2, c, grd, bet2, bet
+  real(rtype) :: f0, f1, f2, f3, f4, f5
   c=7.0_rtype
```

Comment:
It is great that you brought in these variables in this subroutine as they are used locally used here.  Would it make sense to make them `parameters`? I have been doing that after ensuring that they are indeed parameters. It makes it easier to read the code when we know that these variable will not change during the simulation.

Reply:
Done, good suggestion. Thanks.